src/components/buttons/button.tsx

```diff
@@ -1,29 +1,45 @@
 import React from 'react';
 import styled from 'styled-components';
-import { TextStyles } from '../typography/typography-v2';
+import { Icon, IconSize } from '../icon/icon';
+import { resetButton } from '../utils';
+import { lighten } from 'polished';
 
 const StyledButton = styled.button`
-  height: 40px;
+  ${resetButton};
+  color: #ffffff;
+  font-size: 16px;
+  font-weight: bold;
+  line-height: 40px;
+  border-radius: 30px;
+  vertical-align: middle;
+  cursor: pointer;
 
   background: linear-gradient(275.41deg, #543fd7 0%, #2756fd 100%);
-  border-radius: 30px;
-  border: none;
 
   &:hover {
     background: #3e61ee;
   }
+
+  &:active {
+    background: ${lighten(0.1, '#3e61ee')};
+  }
 `;
 
-const StyledIcon = styled.span`
-  padding-left: 24px;
-  padding-right: 16px;
+const StyledIcon = styled(Icon)`
+  padding-right: 10px;
+  // pull the fixed size icon closer to the text
+  margin-left: -4px;
+  /**
+    This is a magic number to match the vertical middle of the button
+    with the icon which has a fixed bounding box which won't allow us to align
+    it based on the actual shape of the icon.
+   */
+  vertical-align: -18%;
```

Comment:
Same here, the bounding box doesn't allow us to let the shape of the icon "flow" with the text. If we had a proper shape without huge amounts of spare space we could use the different modes of vertical-align. Icon fonts can do that because we basically have no bounding box per se (but usually they get one from the css of the icon framework)